src/cpp/rtps/transport/UDPSenderResource.hpp

```diff
@@ -43,14 +43,26 @@ class UDPSenderResource : public fastrtps::rtps::SenderResource
                     transport.CloseOutputChannel(socket_);
                 };
 
-        send_lambda_ = [this, &transport](
-            const fastrtps::rtps::octet* data,
-            uint32_t dataSize,
+        send_buffers_lambda_ = [this, &transport](
+            const NetworkBuffer* buffers,
+            size_t num_buffers,
+            uint32_t total_bytes,
             fastrtps::rtps::LocatorsIterator* destination_locators_begin,
             fastrtps::rtps::LocatorsIterator* destination_locators_end,
             const std::chrono::steady_clock::time_point& max_blocking_time_point) -> bool
                 {
-                    return transport.send(data, dataSize, socket_, destination_locators_begin,
+                    assert(num_buffers < 4);
+
+                    std::array<asio::const_buffer, 3> asio_buffers;
+                    uint32_t total_size = 0;
```

Comment:
```suggestion
                    size_t total_size = 0;
```

Reply:
I have renamed this to `total_num_bytes`

Reply:
See f5a9a8fc3b28de332619b6c2a40dbcbf48a7f4a7